src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1055,11 +1316,10 @@ static private ErrorCode GetThisLvalueError(BindValueKind kind)
 
                 case BindValueKind.IncrementDecrement:
                     return ErrorCode.ERR_IncrementLvalueExpected;
-            }
 
-            if (RequiresReturnableReference(kind))
-            {
-                return ErrorCode.ERR_RefReturnStructThis;
+                case BindValueKind.RefReturn:
+                case BindValueKind.ReadonlyRef:
+                    return ErrorCode.ERR_RefReturnStructThis;
```

Comment:
Error code ERR_RefReturnStructThis is used in two places. Consider using a different error code here.

Reply:
So far I find the error message adequate for both situations. In either case we cannot continue because we need to return "this" by reference, just one case is more direct than another. If I make a different error it would be saying essentially the same thing. I will keep it as-is. 

---
In reply to: [135401375](https://github.com/dotnet/roslyn/pull/21334#discussion_r135401375) [](ancestors = 135401375)